clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManager.java

```diff
@@ -78,14 +78,14 @@ public interface MembershipManager extends RequestManager {
     /**
      * Notify the member that an error heartbeat response was received.
      */
-    void onHeartbeatFailure();
+    void onHeartbeatFailure(boolean retriable);
```

Comment:
From my understanding of the code, the `retriable` flag here only affects whether or not we update metrics. Is that correct? I'm not suggesting we change the flag name, it's just a little confusing (to me) from an API standpoint. Perhaps some JavaDoc to explain what behavior change that flag affects?

Reply:
the change is that `onHeartbeatFailure` is now called for all failures (retriable and non-retriable), so seemed sensible to have a param indicating the nature of the failure because different actions may be required depending on it. It's not specific to metrics, even though that's the only action done now on non-retriable.  

I will add a param description here along the lines of "retriable -> True if the heartbeat request failed with a retriable error". Nothing specific to the metrics, it feels its very specific to the impl, and clear enough from it (and would have to be maintained the moment we come add more action to the `onHeartbeatFailure`), makes sense?